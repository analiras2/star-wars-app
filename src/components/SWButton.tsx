import React, { PureComponent } from "react";
import { TouchableOpacity } from "react-native";
import styles from "../res/styles";
import { colors } from "../res/colors";
import SWText from "./SWText";

interface Props {
  title: string;
  onPress(): void;
  color?: string;
  margin?: number;
  vertical?: number;
  horizontal?: number;
  top?: number;
  bottom?: number;
  right?: number;
  left?: number;
}
export default class SWButton extends PureComponent<Props> {
  render() {
    return (
      <TouchableOpacity
        activeOpacity={0.6}
        style={styles({ ...this.props }).swButton}
      >
        <SWText
          title={this.props.title.toUpperCase()}
          textAlign="center"
          color={colors.primary}
          medium
          bold
        />
      </TouchableOpacity>
    );
  }
}
