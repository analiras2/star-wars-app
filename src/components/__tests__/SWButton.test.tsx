import { shallow } from "enzyme";
import React from "react";
import { render } from "react-native-testing-library";
import renderer from "react-test-renderer";
import SWButton from "../SWButton";

describe("SWButton", () => {
  const props = { title: "Star Wars", onPress: () => {} };
  const wrapper = shallow(<SWButton {...props} />);
  const { getByText } = render(<SWButton {...props} />);

  describe("validating SWButton texts", () => {
    it("should render a Star Wars", () => {
      expect(getByText(/Star Wars/i)).toBeDefined();
    });
  });

  describe("rendering Components", () => {
    it("should render a <TouchableOpacity />", () => {
      expect(wrapper.find("TouchableOpacity")).toHaveLength(1);
    });

    it("should render a <SWText />", () => {
      expect(wrapper.find("SWText")).toHaveLength(1);
    });
  });

  it("renders SWButton correctly", () => {
    const tree = renderer.create(<SWButton {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
