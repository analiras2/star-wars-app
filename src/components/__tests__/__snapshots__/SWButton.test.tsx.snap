// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SWButton renders SWButton correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": undefined,
      "borderRadius": 8,
      "height": 40,
      "justifyContent": "center",
      "marginBottom": 0,
      "marginLeft": 0,
      "marginRight": 0,
      "marginTop": 0,
      "opacity": 1,
      "paddingHorizontal": 20,
      "width": "90%",
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "auto",
        "color": "#000",
        "flex": 0,
        "fontFamily": "DIN-Bold",
        "fontSize": 18,
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "textAlign": "center",
      }
    }
  >
    STAR WARS
  </Text>
</View>
`;
